app/aws/s3.py

```diff
@@ -119,8 +119,12 @@ def get_list_of_files_by_suffix(bucket_name, subfolder='', suffix='', last_modif
     )
 
     for page in page_iterator:
-        for obj in page['Contents']:
-            key = obj['Key'].lower()
-            if key.endswith(suffix.lower()):
-                if not last_modified or obj['LastModified'] >= last_modified:
-                    yield key
+        try:
```

Comment:
Can you query without exceptioning?

Exceptions are for exceptional circumstances this seems like a normal event therefore should be handled by logic?

Reply:
Changed to the query Leo suggested.